Ecal/include/Ecal/EcalMipTrackingProcessor.h

```diff
@@ -0,0 +1,147 @@
+/**
+ * @file EcalMipTrackingProcessor.h
+ * @brief Class that determines MIP tracking information using ECAL hit
+ * information
+ * @author Owen Colegrove, Danyi Zhang, Jihoon Yoo, Tamas Vami (UCSB)
+ */
+
+#ifndef EVENTPROC_ECALMIPTRACKINGPROCESSOR_H_
+#define EVENTPROC_ECALMIPTRACKINGPROCESSOR_H_
+
+// LDMX
+#include "DetDescr/EcalGeometry.h"
+#include "DetDescr/EcalID.h"
+#include "Ecal/Event/EcalMipResult.h"
+#include "Ecal/Event/EcalTrajectoryInfo.h"
+#include "Ecal/Event/EcalVetoResult.h"
+#include "Framework/Configure/Parameters.h"
+#include "Framework/EventProcessor.h"
+
+// C++ Standard Library
+#include <algorithm>
+#include <chrono>
+#include <cmath>
+#include <iomanip>
+#include <map>
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+// ROOT
+#include "TDecompSVD.h"
+#include "TMatrixD.h"
+#include "TVector3.h"
+
+namespace ecal {
+class EcalMipTrackingProcessor : public framework::Producer {
+ public:
+  typedef std::pair<ldmx::EcalID, float> CellEnergyPair;
+
+  typedef std::pair<float, float> XYCoords;
```

Comment:
No reason to repeat yourself, we can pull in the type from the event object.
```suggestion
using XYCoords = ldmx::XYCoords;
```